client-core/src/service/key_service.rs

```diff
@@ -239,12 +195,18 @@ where
         let cointype = get_bip44_coin_type();
         log::debug!("coin type={}", cointype);
         let account = if is_staking { 1 } else { 0 };
-        let extended = ExtendedPrivKey::derive(
-            &seed_bytes.expect("generate_keypair_hd get seed bytes"),
-            format!("m/44'/{}'/{}'/0/{}", cointype, account, index).as_str(),
-        )
-        .map_err(|_e| Error::new(ErrorKind::InvalidInput, "hdwallet derive new address"))?;
-        let mut secret_key_bytes = extended.secret();
+
+        let chain_path = format!("m/44'/{}'/{}'/0/{}", cointype, account, index);
+        let key_chain = DefaultKeyChain::new(
+            ExtendedPrivKey::with_seed(&seed_bytes.as_ref().expect("hdwallet get extended")[..])
+                .map_err(|_e| Error::new(ErrorKind::InvalidInput, "invalid seed bytes"))?,
+        );
+        let (key, _derivation) = key_chain
+            .derive_private_key(ChainPath::from(chain_path.to_string()))
+            .map_err(|_e| Error::new(ErrorKind::InvalidInput, "hdwallet derive private key"))?;
```

Comment:
Can you use `chain()` instead of `map_err()`?

Reply:
trait bounds doen't match by Send+Sync